akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -317,6 +318,96 @@ class CircuitBreaker(
   /**
+   * Adds a callback to execute when call finished with success.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Long ⇒ Unit): CircuitBreaker = onCallSuccess(new Consumer[Long] {
+    def accept(result: Long): Unit = callback(result)
+  })
+
+  /**
+   * JavaAPI for onCallSuccess
+   *
+   * @param callback Handler to be invoked on successful call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallSuccess(callback: Consumer[Long]): CircuitBreaker = {
+    successListeners add callback
+    this
+  }
+
+  /**
+   * Adds a callback to execute when call finished with failure.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on failed call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallFailure(callback: Long ⇒ Unit): CircuitBreaker = onCallFailure(new Consumer[Long] {
+    def accept(result: Long): Unit = callback(result)
+  })
+
+  /**
+   * JavaAPI for onCallFailure
+   *
+   * @param callback Handler to be invoked on failed call, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallFailure(callback: Consumer[Long]): CircuitBreaker = {
+    callFailureListeners add callback
+    this
+  }
+
+  /**
+   * Adds a callback to execute when call finished with timeout.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on call finished with timeout, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallTimeout(callback: Long ⇒ Unit): CircuitBreaker = onCallTimeout(new Consumer[Long] {
+    def accept(result: Long): Unit = callback(result)
+  })
+
+  /**
+   * JavaAPI for onCallTimeout
+   *
+   * @param callback Handler to be invoked on call finished with timeout, where passed value is elapsed time in nanoseconds.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallTimeout(callback: Consumer[Long]): CircuitBreaker = {
+    callTimeoutListeners add callback
+    this
+  }
+
+  /**
+   * Adds a callback to execute when call was failed due to open breaker.
+   *
+   * The callback is run in the [[scala.concurrent.ExecutionContext]] supplied in the constructor.
+   *
+   * @param callback Handler to be invoked on call failed due to open breaker.
+   * @return CircuitBreaker for fluent usage
+   */
+  def onCallBreakerOpen(callback: ⇒ Unit): CircuitBreaker = onCallBreakerOpen(new Runnable { def run = callback })
```

Comment:
onOpen is invoked when CB goes into open state, onCallBreakerOpen is invoked when call was made when CB is in open state.